test/client/feature/geojson-external-instance.spec.js

```diff
@@ -0,0 +1,322 @@
+
+        it('prioritizes top level id', async () => {
+            const result = await loadGeoJSONExternalInstance(
+                'feature-collection-id-twice.geojson'
+            );
+
+            expect(
+                result.documentElement.children[0].childElementCount
+            ).to.equal(4);
+            expect(
+                result.documentElement.children[0].querySelector('id')
+                    .textContent
+            ).to.equal('top-level-id');
+        });
+
+        it('ignores unknown top level properties', async () => {
+            const result = await loadGeoJSONExternalInstance(
+                'feature-collection-extra-feature-toplevel.geojson'
+            );
+
+            expect(
+                result.documentElement.children[0].childElementCount
+            ).to.equal(3);
+            expect(
+                result.documentElement.children[0].querySelector('ignored')
+            ).to.equal(null);
+        });
+
+        it('adds features with no properties', async () => {
+            const result = await loadGeoJSONExternalInstance(
+                'feature-collection-no-properties.geojson'
+            );
+
+            expect(
+                result.documentElement.children[0].childElementCount
+            ).to.equal(1);
+        });
+
+        it('allows null feature property values', async () => {
+            const result = await loadGeoJSONExternalInstance(
+                'feature-collection-with-null.geojson'
+            );
+
+            expect(
+                result.documentElement.children[0].childElementCount
+            ).to.equal(4);
+            expect(
+                result.documentElement.children[0].querySelector('extra')
+                    .textContent
+            ).to.equal('');
+        });
+    });
+
+    describe('integration', () => {
+        const basePath = '-';
+        const serverURL = 'example.com';
+        const enketoId = 'external-select-geojson.xml';
+
+        // Copied from current transformation result... there's probably a better way?
+        const transformed = {
+            form: '<form autocomplete="off" novalidate="novalidate" class="or clearfix" dir="ltr" data-form-id="external-select-xml-3">\n<!--This form was created by transforming an ODK/OpenRosa-flavored (X)Form using an XSL stylesheet created by Enketo LLC.--><section class="form-logo"></section><h3 dir="auto" id="form-title">GeoJSON External Secondary Instance 3</h3>\n    \n    \n        <fieldset class="question simple-select "><fieldset><legend><span lang="" class="question-label active">Question</span>\n                    </legend><div class="option-wrapper"><label class="itemset-template" data-items-path="instance(\'external-geojson\')/root/item"><input type="radio" name="/data/q" data-name="/data/q" data-type-xml="string" value=""></label><span class="itemset-labels" data-value-ref="id" data-label-ref="name">\n            </span></div></fieldset></fieldset>\n    \n</form>',
```

Comment:
This scares me? But I guess we can deal with it if ever it fails!

Reply:
Yeah, scares me too. Worth noting, this is also how many tests are written for `connection.js`. We should really consider setting up a real transformer service for client integration tests to use.